api/tests/routes/public/collective/endpoints/patch_collective_offer_public_provider_test.py

```diff
@@ -123,6 +123,33 @@ def test_patch_offer(self, client):
         assert offer.institutionId == educational_institution.id
         assert educational_institution.isActive is True
 
+    @override_features(ENABLE_PROVIDER_AUTHENTIFICATION=True)
+    def test_patch_offer_price_should_be_lower(self, client):
+        # Given
+        offerer = offerers_factories.OffererFactory()
+        offerers_factories.UserOffererFactory(offerer=offerer)
+        offerers_factories.ApiKeyFactory(offerer=offerer)
+        venue = offerers_factories.VenueFactory(managingOfferer=offerer)
+        stock = educational_factories.CollectiveStockFactory(
+            collectiveOffer__isPublicApi=True,
+            collectiveOffer__imageCredit="pouet",
+            collectiveOffer__imageId="123456789",
+            collectiveOffer__venue=venue,
+        )
```

Comment:
En plus court : 

```suggestion
        stock = educational_factories.CollectiveStockFactory()
        offerer = stock.collectiveOffer.venue.managingOfferer
        offerers_factories.ApiKeyFactory(offerer=offerer)
```

Il est important que chaque test ne créé que les données qui sont strictement nécessaires. Cela facilite grandement la lecture et la maintenance des tests.

Idem dans l'autre module de test. D'ailleurs, la raison pour laquelle on double tous les tests dans ces 2 modules m'échappe... On ne change que le feature flag `ENABLE_PROVIDER_AUTHENTIFICATION`, mais sans changer la façon dont on s'authentifie ni les données de base du test. Que cherche-t-on à vérifier de différent dans ces 2 tests ?

Reply:
a l'origine seul le test avec l'offerer existait, mais un ticket demandant de changer les offerers par les providers a été fait précédemment avec un feature flag dans le but de faire une transition, bientôt, la partie avec les offerers ne sera plus active, j'attend de savoir quand pour effacer ces tests.